integration/test_backend_configs.py

```diff
@@ -0,0 +1,67 @@
+"""Tests that we can use all available backend configs."""
+
+from datetime import datetime
+
+from pytket.backends.backendresult import BackendResult
+from pytket.circuit import Circuit
+
+import qnexus as qnx
+
+CONFIGS_REQUIRE_NO_MEASURE = [qnx.AerUnitaryConfig]
+CONFIGS_NOT_TO_EXECUTE = [
+    # We won't execute for these configs as they run on quantum hardware
+    qnx.IBMQConfig
+]
+
+
+def test_basic_backend_config_usage(
+    _authenticated_nexus: None,
+    backend_config: qnx.BackendConfig,
+    qa_project_name: str,
+) -> None:
+    """Test basic functionality of supported BackendConfigs."""
+
+    project_ref = qnx.projects.get(name_like=qa_project_name)
+
+    my_circ = Circuit(2, 2).H(0).CX(0, 1)
+
+    if backend_config.__class__ in CONFIGS_REQUIRE_NO_MEASURE:
+        my_circ.measure_all()
+
+    my_circ = qnx.circuits.upload(
+        circuit=my_circ,
+        name=f"QA BackendConfig circuit {backend_config.__class__.__name__}_{datetime.now()}",
+        description=f"This can be safely deleted. Test Run: {datetime.now()}",
+        project=project_ref,
+    )
+
+    compile_job_ref = qnx.start_compile_job(
+        programs=[my_circ],
+        name=f"QA BackendConfig compile job {backend_config.__class__.__name__}_{datetime.now()}",
+        optimisation_level=0,
+        backend_config=backend_config,
+        project=project_ref,
+    )
+
+    qnx.jobs.wait_for(compile_job_ref)
+
+    # Check the backend config as stored in Nexus matches the one specified
+    assert qnx.jobs.get(id=compile_job_ref.id).backend_config == backend_config
+
+    if backend_config.__class__ in CONFIGS_NOT_TO_EXECUTE:
+        return
+
+    execute_job_ref = qnx.start_execute_job(
+        programs=[item.get_output() for item in qnx.jobs.results(compile_job_ref)],
+        name=f"Execute {backend_config.__class__.__name__}_{datetime.now()}",
+        n_shots=[100],
+        backend_config=backend_config,
+        project=project_ref,
+    )
+
+    qnx.jobs.wait_for(execute_job_ref)
```

Comment:
In case of the IBMQ devices, I don't think we want to wait for the remote jobs to be processed. Maybe we can just assert that the jobs are scheduled/queued remotely, and then cancel them?